packages/browser-wallet/src/popup/pages/AddTokens/ExternalAddTokens.tsx

```diff
@@ -0,0 +1,237 @@
+        return null;
+    }
+
+    const updateTokenStatus = (index: number) => {
+        const token = addingTokens[index];
+        const status = token.status === ChoiceStatus.chosen ? ChoiceStatus.discarded : ChoiceStatus.chosen;
+        addingTokens[index] = { ...token, status };
+        setAddingTokens([...addingTokens]);
+    };
+
+    const allExisting = addingTokens.every(({ status }) => status === ChoiceStatus.existing);
+
+    return (
+        <ExternalRequestLayout>
+            <ConnectedBox accountAddress={account} url={new URL(url).origin} />
+            <div className="h-full flex-column align-center">
+                <p>
+                    {t(allExisting ? 'prompt.descriptionAllExisting' : 'prompt.description', { dApp: displayUrl(url) })}
+                </p>
+                <UncontrolledInput
+                    readOnly
+                    className="add-tokens-prompt__input w-full"
+                    label={t('contractName')}
+                    value={contractDetails.contractName}
+                />
+                <div className="add-tokens-prompt__token-container w-full">
+                    {addingTokens.map((token, index) => (
+                        <DisplayToken
+                            key={token.id}
+                            status={token.status}
+                            metadata={token.metadata}
+                            balance={0n}
+                            updateStatus={() => updateTokenStatus(index)}
+                        />
+                    ))}
+                </div>
+                <div className="flex p-b-10  m-t-auto">
+                    {allExisting || (
+                        <>
+                            <Button
+                                width="narrow"
+                                className="m-r-10"
+                                onClick={withClose(() => respond(addingTokens.filter(onlyExisting).map(getId)))}
+                            >
+                                {t('prompt.reject')}
+                            </Button>
+                            <Button
+                                width="narrow"
+                                onClick={() =>
+                                    onFinish(
+                                        addingTokens
+                                            .filter((token) => token.status !== ChoiceStatus.discarded)
+                                            .map(({ status, ...token }) => token)
+                                    )
+                                }
+                            >
+                                {t('prompt.add')}
+                            </Button>
+                        </>
+                    )}
+                    {allExisting && (
+                        <Button width="wide" onClick={withClose(() => respond(addingTokens.map(getId)))}>
+                            {t('prompt.finish')}
+                        </Button>
+                    )}
```

Comment:
Won't the user want another way out, even if the tokens are already added? You reveal that you have an interest in the tokens given in the request.

Reply:
They can close the window, I guess :thinking:

Reply:
Yes.. Is that fine? 🤷

Reply:
I'd think so? :shrug: